node/wrappers.js

```diff
@@ -21,9 +21,10 @@ exports.addWrapper = addWrapper;
 function createWrapperManager(object, key) {
   const func = object[key];
   if (! isManaged(func)) {
-    (object[key] = function(param, filename) {
+    (object[key] = function wrapperFunc(param, filename) {
       const pkgInfo = utils.getPkgInfo(path.dirname(filename));
-      const wrapper = pkgInfo === null ? null : findWrapper(object[key], pkgInfo.range);
+      const wrapper = pkgInfo === null ? null :
+        findWrapper(wrapperFunc, pkgInfo.range);
 
```

Comment:
The problem with using `wrapperFunc` as the anchor is once the `_extensions[".js"]` is wrapped by another we'll stop getting new entries added to `wrapperFunc` and they'll be added to the new wrapped func as we'll detect that it's no longer managed by us.

Instead we could use a symbol on the `_extensions` object to store the metadata which should persist through the hooks, `'.js'` and friends, being wrapped and rewrapped.